src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,256 @@
+from astropy.stats import sigma_clipped_stats
+
+from ctapipe.containers import StatisticsContainer
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import (
+    CaselessStrEnum,
+    Int,
+    List,
+)
+
+
+class StatisticsExtractor(TelescopeComponent):
+    """
+    Base component to handle the extraction of the statistics from a dl1 table
+    containing charges, peak times and/or charge variances (images).
+    """
+
+    chunk_size = Int(
+        2500,
+        help="Size of the chunk used for the calculation of the statistical values",
+    ).tag(config=True)
+
+    def __call__(
+        self,
+        dl1_table,
+        masked_pixels_of_sample=None,
+        chunk_shift=None,
+        col_name="image",
+    ) -> list:
+        """
+        Prepare the extraction chunks and call the relevant function of the particular extractor
+        to extract the statistical values.
+
+        Parameters
+        ----------
+        dl1_table : astropy.table.Table
+            DL1 table with images of shape (n_images, n_channels, n_pix)
+            and timestamps of shape (n_images, ) stored in an astropy Table
+        masked_pixels_of_sample : ndarray
+            boolean array of masked pixels of shape (n_pix, ) that are not available for processing
+        chunk_shift : int
+            number of samples to shift the extraction chunk
+        col_name : string
+            column name in the DL1 table
+
+        Returns
+        -------
+        List StatisticsContainer:
+            List of extracted statistics and extraction chunks
+        """
+
+        # Check if the length of the dl1 table is greater or equal than the size of the chunk.
+        if len(dl1_table[col_name]) < self.chunk_size:
+            raise ValueError(
+                f"The length of the DL1 table ({len(dl1_table[col_name])}) must be greater or equal than the size of the chunk ({self.chunk_size})."
+            )
+
+        # Function to split table data into appropriated chunks
+        def _get_chunks(dl1_table_data, chunk_shift):
+            if chunk_shift is None:
+                return (
+                    (
+                        dl1_table_data[i : i + self.chunk_size]
+                        if i + self.chunk_size <= len(dl1_table_data)
+                        else dl1_table_data[
+                            len(dl1_table_data) - self.chunk_size : len(dl1_table_data)
+                        ]
+                    )
+                    for i in range(0, len(dl1_table_data), self.chunk_size)
+                )
+            else:
+                return (
+                    dl1_table_data[i : i + self.chunk_size]
+                    for i in range(
+                        0, len(dl1_table_data) - self.chunk_size, chunk_shift
+                    )
+                )
+
+        # Get the chunks for the timestamps and selected column name
+        time_chunks = _get_chunks(dl1_table["time_mono"], chunk_shift)
```

Comment:
Why do we loop over two chunk iterators instead of just selecting the rows of the whole table?

Reply:
Good point!